tests/test_views.py

```diff
@@ -0,0 +1,756 @@
+        response = views.app_event(request)
+        assert response.status_code == 200
+
+    def test_list_number_of_events(self, rf):
+        factories.EventFactory.create_batch(self.RESULTS_PER_PAGE * 2)
+        request = rf.get('/')
+        response = views.app_event(request)
+        xml = objectify.fromstring(response.content)
+
+        assert len(xml.entry) == self.RESULTS_PER_PAGE
+
+    @pytest.mark.xfail(reason='Global name DATE_FORMAT is not defined.')
+    def test_list_filtering_by_start_date(self, rf):
+        datetime_obj = timezone.now().replace(2015, 1, 1)
+        factories.EventFactory.create_batch(30, event_date_time=datetime_obj)
+        event = factories.EventFactory.create(event_date_time=timezone.now())
+
+        request = rf.get('/?start_date=01/31/2015')
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
+    @pytest.mark.xfail(reason='Global name DATE_FORMAT is not defined.')
+    def test_list_filtering_by_end_date(self, rf):
+        datetime_obj = timezone.now().replace(2015, 1, 1)
+        factories.EventFactory.create_batch(30, event_date_time=timezone.now())
+        event = factories.EventFactory.create(event_date_time=datetime_obj)
+
+        request = rf.get('/?end_date=01/31/2015')
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
+    def test_list_filtering_by_linking_object_id(self, rf):
+        factories.EventFactory.create_batch(30)
+        event = factories.EventFactory.create(linking_objects=True)
+        linking_object = event.linking_objects.first()
+
+        url = '/?link_object_id={0}'.format(linking_object.object_identifier)
+        request = rf.get(url)
+        response = views.app_event(request)
+
+        assert self.event_in_filtered_results(response, event)
+
+    def test_list_filtering_by_event_outcome(self, rf):
+        factories.EventFactory.create_batch(30)
+
+        event_outcome = 'success'
+        event = factories.EventFactory.create(event_outcome=event_outcome)
+
+        request = rf.get('?outcome={0}'.format(event_outcome))
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
+    def test_list_filtering_by_event_type(self, rf):
+        factories.EventFactory.create_batch(30)
+
+        event_type = 'Test Type'
+        event = factories.EventFactory.create(event_type=event_type)
+
+        request = rf.get('?type={0}'.format(event_type))
+        response = views.app_event(request)
+        assert self.event_in_filtered_results(response, event)
+
+    def test_get_with_identifier_returns_ok(self, rf):
+        event = factories.EventFactory.create()
+        request = rf.get('/', HTTP_HOST='example.com')
+        response = views.app_event(request, event.event_identifier)
+        assert response.status_code == 200
+
+    def test_get_with_identifier_returns_not_found(self, rf):
```

Comment:
This one is a little different. In the `app_agent` view, we were getting the object first, then checking the request methods. In `app_event`, the request method is checked, then the object is fetched. Thus, in `app_event` we are allowing the `Http404` exception to be raised in multiple places.

I will say this though. I think the names need to be modified.
- `test_get_with_invalid_identifier_returns_not_found`
- `test_delete_with_invalid_identifier_returns_not_found`